plugin/CactbotEventSource/FateWatcher.cs

```diff
@@ -7,97 +7,73 @@
 using System.Runtime.InteropServices;
 using Newtonsoft.Json.Linq;
 using System.Threading;
+using System.Net.Http;
+using Newtonsoft.Json;
 
 namespace Cactbot {
+  //Converts uints to hex formatted strings and vice versa
+  public sealed class HexStringJsonConverter : JsonConverter
+  {
+    public override bool CanConvert(Type objectType)
+    {
+      return typeof(uint).Equals(objectType);
+    }
+
+    public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+    {
+      writer.WriteValue($"0x{value:x}");
+    }
+
+    public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+    {
+      if (reader.ValueType.FullName == typeof(string).FullName)
+      {
+        string str = (string)reader.Value;
+        if (str == null || !str.StartsWith("0x"))
+          throw new JsonSerializationException();
+        return Convert.ToUInt32(str.Substring("0x".Length), 16);
+      }
+      else
+        throw new JsonSerializationException();
+    }
+  }
+
   public class FateWatcher {
     private CactbotEventSource client_;
-    private string region_;
     private IDataSubscription subscription;
 
-    // Fate start
-    // param1: fateID
-    // param2: unknown
-    //
-    // Fate end
-    // param1: fateID
-    //
-    // Fate update
-    // param1: fateID
-    // param2: progress (0-100)
-    private struct AC143OPCodes {
-      public AC143OPCodes(int add_, int remove_, int update_) { this.add = add_; this.remove = remove_; this.update = update_; }
-      public int add;
-      public int remove;
-      public int update;
-    };
-    private static readonly AC143OPCodes ac143_v5_2 = new AC143OPCodes(
-      0x935,
-      0x936,
-      0x93E
-    );
-
-    private struct CEDirectorOPCodes {
-      public CEDirectorOPCodes(int size_, int opcode_) { this.size = size_; this.opcode = opcode_; }
-      public int size;
-      public int opcode;
+    private static readonly string remoteOpcodeURL = "https://katttails.github.io/TestRepo/FATEOpcodes.json";
```

Comment:
TODO: This needs adjusting before the final merge based on where exactly we're going to be putting the json.